### PR TITLE
🐛 Sync nodes when patching

### PR DESCRIPTION
**Pull request by giancarloromeo**
## What do these changes do?
Whenever a node is _patched_ - in particular `inputs` and/or `outputs` get modified - all the nodes in the project need to be checked for state changes.

The issue was more evident trying to reproduce [#6505](https://github.com/ITISFoundation/osparc-simcore/issues/6505).

## Related issue/s

- Resolves: https://github.com/ITISFoundation/osparc-simcore/issues/6505

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py [hunk 1]
```diff
@@ -1047,13 +1047,17 @@ async def patch_project_node(
     if _node_patch_exclude_unset.get("label"):
         await dynamic_scheduler_api.update_projects_networks(app, project_id=project_id)
 
-    # 5. Updates project states for user, if inputs have been changed
-    if "inputs" in _node_patch_exclude_unset:
+    # 5. Updates project states for user, if inputs/outputs have been changed
+    if {"inputs", "outputs"} & _node_patch_exclude_unset.keys():
```
Comment (matusdrobuliak66) [inline services/web/server/src/simcore_service_webserver/projects/projects_api.py:1051]: Cool, thanks! Maybe we can extend this test with these use cases?
`osparc-simcore/services/web/server/tests/unit/with_dbs/02/test_projects_nodes_handlers__patch.py`

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py (hunk 1 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/projects/projects_api.py:1051]: i agree with @matusdrobuliak66 . A test is necessary here. Either extend or a new one.

Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/projects_api.py:1051]: `a & b` with `a,b` being sets, is an intersection?   I would use `a.intersects(b)`

**Review by pcrespov — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py (hunk 1 above)
Comment (giancarloromeo) [reply services/web/server/src/simcore_service_webserver/projects/projects_api.py:1051]: Yup, it's intersection https://docs.python.org/3.10/library/stdtypes.html#frozenset.intersection

**Review by sanderegg — APPROVED** [review]
great! thanks! don't forget to add in your case:
- resolves CASE_LINK or fixes CASE_LINK,
This will auto fix it and link the PR to the issue automatically

**Review by GitHK — COMMENTED** [review]
would also appreciate a test.

**Pull request merged**